src/snowflake/snowpark/modin/plugin/_internal/aggregation_utils.py

```diff
@@ -1127,11 +1128,71 @@ def using_named_aggregations_for_func(func: Any) -> bool:
     )
 
 
-def format_kwargs_for_error_message(kwargs: dict[Any, Any]) -> str:
+def repr_aggregate_function(agg_func: AggFuncType, agg_kwargs: Mapping) -> str:
     """
-    Helper method to format a kwargs dictionary for an error message.
+    Represent an aggregation function as a string.
 
-    Returns a string containing the keys + values of kwargs formatted like so:
-    "key1=value1, key2=value2, ..."
+    Use this function to represent aggregation functions in error message to
+    the user. This user will hide sensitive information, like axis labels or
+    names of callables, in the function description.
+
+    Args:
+        agg_func: AggFuncType
+            The aggregation function from the user. This may be a list-like or a
+            dictionary containing multiple aggregations.
+        agg_kwargs: Mapping
+            The keyword arguments for the aggregation function.
+
+    Returns:
+        str
+            The representation of the aggregation function.
     """
-    return ", ".join([f"{key}={value}" for key, value in kwargs.items()])
+    if using_named_aggregations_for_func(agg_func):
+        # New axis labels are sensitive, so replace them with "new_label."
+        # Existing axis labels are sensitive, so replace them with "label."
+        return ", ".join(
+            f"new_label=(label, {repr_aggregate_function(f, agg_kwargs)})"
+            for _, f in agg_kwargs.values()
+        )
+    if isinstance(agg_func, str):
+        # Strings functions represent names of pandas functions, e.g.
+        # "sum" means to aggregate with pandas.Series.sum. string function
+        # identifiers are not sensitive.
+        return repr(agg_func)
+    if is_dict_like(agg_func):
+        # axis labels in the dictionary keys are sensitive, so replace them with
+        # "label."
+        return (
+            "{"
+            + ", ".join(
+                f"label: {repr_aggregate_function(agg_func[key], agg_kwargs)}"
+                for key in agg_func.keys()
+            )
+            + "}"
+        )
+    if is_list_like(agg_func):
+        return f"[{', '.join(repr_aggregate_function(func, agg_kwargs) for func in agg_func)}]"
+    if isinstance(agg_func, BuiltinFunctionType):
+        return repr(agg_func)
+
+    # for built-in classes like `list`, return "list" as opposed to repr(list),
```

Comment:
```suggestion
    # if agg_func is a callable version of a built-in classes like `list`, return "list" as opposed to repr(list),
```
makes things a little clearer?